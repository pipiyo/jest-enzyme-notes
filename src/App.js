import React, { Component } from 'react';
import logo from './logo.svg';
import './App.css';

class App extends Component {
  state = { 
    on: false,
    input: '',
    mainColor: 'blue',
    lifeCycle: ''
  }
  componentDidMount() {
   this.setState({ lifeCycle: 'componentDidMount' })
  }
  componentWillReceiveProps() {
    this.setState({ lifeCycle: 'componentWillReceiveProps' })
  }
  render() {
    return (
      <div className="App">
        <header className="App-header">
          <img src={logo} className="App-logo" alt="logo" />
          <p>
            Edit <code>src/App.js</code> and save to reload.
          </p>
          <h1 className="App-title">Welcome to React</h1> 
          <h3 className={this.state.mainColor}>Everyone is Welcome!</h3>         
          <a
            className="App-link"
            href="https://reactjs.org"
            target="_blank"
            rel="noopener noreferrer"
          >
            Learn React
          </a>
        </header>
        <ul className="tyler">
          <li>Test 1</li>
          <li>Test 2</li>
          <li>Test 3</li>
        </ul>
        <p className='button-state'>{this.state.on ? 'Yes!' : 'No!'}</p>
        <button onClick={() => this.setState({on: true})}>Click</button>
        <h2>{this.state.input}</h2>
        <input onChange={(e) => this.setState({input: e.currentTarget.value})} type='text' />
        <p className='lifeCycle'>{this.state.lifeCycle}</p>
      </div>
    );
  }
}

export class Link extends Component {
  render() {
    return this.props.hide ? null : <a href={this.props.address}>Click</a>
  }
}

export default App;
