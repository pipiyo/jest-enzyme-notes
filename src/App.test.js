import React from 'react';
import ReactDOM from 'react-dom';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';

import App, { Link } from './App';

configure({ adapter: new Adapter() })

describe('<App />', () => {
  	const wrapper = shallow(<App />)

//  it('should render App', () => {
//    const wrapper = shallow(<App />, {context: {}, disableLifecycleMethods: false})
//    console.log(wrapper.debug())
//  })

  it('should contain 1 p element', () => {
    expect(wrapper.find('p').length).toBe(1)
  })

  it('should ul contain 3 li element', () => {
	expect(wrapper.find('ul').children().length).toBe(3)
  })

  it('ul should has class tyler', () => {
	expect(wrapper.find('ul').hasClass('tyler')).toBe(true)
  })  

  it('h1 text should be "Welcome to React"', () => {
	expect(wrapper.find('h1').text()).toBe('Welcome to React')
  })

  it('matches the snapshot', () => {
    const tree = shallow(<App />)
    expect(toJson(tree)).toMatchSnapshot()
  })

})

describe('<Link />', () => {
  it('link component accepts address prop', () => {
    const wrapper = shallow(<Link address='www.google.com' />)
    expect(wrapper.instance().props.address).toBe('www.google.com')
  })
  it('a tag node renders href correctly', () => {
    const wrapper = shallow(<Link address='www.google.com' />)
    expect(wrapper.props().href).toBe('www.google.com')
  })
  it('returns null with true hide prop', () => {
    const wrapper = shallow(<Link hide={false} />)
    expect(wrapper.find('a').length).toBe(1)
    wrapper.setProps({ hide: true })
    expect(wrapper.get(0)).toBeNull()
  })
})