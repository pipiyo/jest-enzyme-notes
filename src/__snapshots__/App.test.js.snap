// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> matches the snapshot 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <img
      alt="logo"
      className="App-logo"
      src="logo.svg"
    />
    <p>
      Edit 
      <code>
        src/App.js
      </code>
       and save to reload.
    </p>
    <h1
      className="App-title"
    >
      Welcome to React
    </h1>
    <ul
      className="tyler"
    >
      <li>
        Test 1
      </li>
      <li>
        Test 2
      </li>
      <li>
        Test 3
      </li>
    </ul>
    <a
      className="App-link"
      href="https://reactjs.org"
      rel="noopener noreferrer"
      target="_blank"
    >
      Learn React
    </a>
  </header>
</div>
`;
